### PR TITLE
Support logging cron output

### PR DESCRIPTION
**Pull request by treydock**
#### Pull Request (PR) description

Allow logging letsencrypt renewal output.  So far hoping this will be useful when post_commands don't fire, so debug what's going wrong there.

### CODE REVIEW

File: manifests/certonly.pp [hunk 1]
```diff
@@ -58,6 +59,7 @@
   Array[String[1]]                          $environment          = [],
   Boolean                                   $manage_cron          = false,
   Boolean                                   $suppress_cron_output = false,
+  Boolean                                   $log_cron_output      = false,
```
Comment (smortex) [inline manifests/certonly.pp:62]: These two are mutually exclusive.  Maybe add a test condition that raise an error if both or true?  Or maybe replace them with a single setting `Enum['suppress', 'log', 'ignore'] $cron_output = 'ignore'` ?

**Review by smortex — COMMENTED** [review]
Please also consider adding / adjusting tests in spec/defines/letsencrypt_certonly_spec.rb near line 435 to ensure the new behavior does not break in the future.

File: manifests/certonly.pp [hunk 2]
```diff
@@ -182,11 +186,18 @@
     $cron_script_ensure = $ensure ? { 'present' => 'file', default => 'absent' }
     $cron_ensure = $ensure
 
-    if $suppress_cron_output {
-      $croncommand = "${maincommand} > /dev/null 2>&1"
-    } else {
-      $croncommand = $maincommand
+    case $cron_output {
```
Comment (root-expert) [inline manifests/certonly.pp:189]: I would use selector syntax here since it's more clean.

**Review by root-expert — APPROVED** [review]

File: manifests/certonly.pp (hunk 2 above)
Comment (treydock) [reply manifests/certonly.pp]: Made this change.

**Comment by root-expert** [comment]
Please squash your commits before merging. Thanks!

**Comment by treydock** [comment]
Commits squashed

**Review by smortex — APPROVED** [review]
LGTM!

**Pull request merged**